table.go

```diff
@@ -54,25 +54,36 @@ func (e ErrCreateSchemaWriter) Error() string {
 }
 
 type TableConfig struct {
-	schema       *dynparquet.Schema
-	rowGroupSize int
+	schema           *dynparquet.Schema
+	rowGroupSize     int
+	blockReaderLimit int
 }
 
 type TableOption func(*TableConfig) error
 
+// WithRowGroupSize sets the size in number of rows for each row group for parquet files. A <= 0 value indicates no limit.
 func WithRowGroupSize(numRows int) TableOption {
 	return func(config *TableConfig) error {
 		config.rowGroupSize = numRows
 		return nil
 	}
 }
 
+// WithBlockReaderLimit sets the limit of go routines that will be used to read persisted block files. A negative number indicates no limit.
+func WithBlockReaderLimit(n int) TableOption {
+	return func(config *TableConfig) error {
+		config.blockReaderLimit = n
+		return nil
+	}
+}
+
 func NewTableConfig(
 	schema *dynparquet.Schema,
 	options ...TableOption,
 ) *TableConfig {
 	t := &TableConfig{
-		schema: schema,
+		schema:           schema,
+		blockReaderLimit: -1,
```

Comment:
I don't think we want unlimited parallelization by default, since this could cause an unwanted memory blowup due to the responses. I think `runtime.NumCPU` is a better default, and recently [someone suggested using GOMAXPROCS with the automaxprocs package](https://github.com/polarsignals/frostdb/pull/231#issuecomment-1278022860), which I think is even better.

Reply:
Yea, I waffled on a default here since this is paralleling network bound calls VS compute so cpu/procs felts like the incorrect default.